build.gradle.kts

```diff
@@ -10,86 +9,115 @@ description = "A Result monad for modelling success or failure operations."
 }
 
+tasks.withType<DependencyUpdatesTask> {
+    rejectVersionIf {
+        listOf("alpha", "beta", "rc", "cr", "m", "eap", "pr").any {
+            candidate.version.contains(it, ignoreCase = true)
+        }
+    }
+}
+
+val dokka by tasks.existing(DokkaTask::class) {
+    outputFormat = "javadoc"
+    outputDirectory = "$buildDir/docs/javadoc"
+}
+
+val javadocJar by tasks.registering(Jar::class) {
+    group = LifecycleBasePlugin.BUILD_GROUP
+    description = "Assembles a jar archive containing the Javadoc API documentation."
+    archiveClassifier.set("javadoc")
+    dependsOn(dokka)
+    from(dokka.get().outputDirectory)
+}
+
 allprojects {
     repositories {
         mavenCentral()
         jcenter()
         maven("https://dl.bintray.com/kotlin/kotlinx")
     }
-
-    tasks.withType<KotlinCompile> {
-        kotlinOptions {
-            jvmTarget = "1.8"
-            freeCompilerArgs = listOf("-Xuse-experimental=kotlin.contracts.ExperimentalContracts")
-        }
-    }
 }
 
 allOpen {
     annotation("org.openjdk.jmh.annotations.State")
     annotation("org.openjdk.jmh.annotations.BenchmarkMode")
 }
 
-sourceSets.create("benchmark") {
-    java {
-        srcDir("src/benchmarks/kotlin")
-    }
-}
-
-val benchmarkImplementation by configurations
+// creating this sourceSet and specifying jvm { withJava() } will create a
+// "jvmBenchmark" sourceSet we can access from kotlin mpp extension
+sourceSets.create("benchmark")
 
 benchmark {
     targets {
-        register("benchmark")
+        register("jvmBenchmark")
     }
 }
 
-dependencies {
-    implementation(kotlin("stdlib"))
-    testImplementation("junit:junit:4.13")
-    testImplementation(kotlin("test-common"))
-    testImplementation(kotlin("test-annotations-common"))
-    testImplementation(kotlin("test-junit"))
-    testImplementation(kotlin("test"))
+kotlin {
+    jvm {
+        withJava()
+    }
+    sourceSets {
+        val commonMain by getting {
+            dependencies {
+                implementation(kotlin("stdlib-common"))
+            }
+        }
 
-    benchmarkImplementation(sourceSets.main.get().output + sourceSets.main.get().runtimeClasspath)
-    benchmarkImplementation("org.jetbrains.kotlinx:kotlinx.benchmark.runtime-jvm:0.2.0-dev-8")
-}
+        val commonTest by getting {
+            dependencies {
+                implementation(kotlin("test-common"))
+                implementation(kotlin("test-annotations-common"))
+            }
+        }
 
-tasks.withType<DependencyUpdatesTask> {
-    rejectVersionIf {
-        listOf("alpha", "beta", "rc", "cr", "m", "eap", "pr").any {
-            candidate.version.contains(it, ignoreCase = true)
+        val jvmMain by getting {
+            dependencies {
+                implementation(kotlin("stdlib-jdk8"))
+            }
+        }
+
+        val jvmTest by getting {
+            dependencies {
+                implementation(kotlin("test-junit"))
+                implementation(kotlin("test"))
```

Comment:
should this be moved to commonTest?

Reply:
I don't think so, see: https://github.com/michaelbull/kotlin-inline-logger/blob/master/build.gradle.kts#L73